File: selenium-webdriver-python-unittest/pages/login_page.py
from pages.abstract_page import AbstractPage

class LoginPage(AbstractPage):

  def enter_username(self, user):
    self.driver.find_element_by_id("username").send_keys(user)
 
  def enter_password(self, password):
    self.driver.find_element_by_id("password").send_keys(password)

  def click_sign_in(self):
    self.driver.find_element_by_xpath("//input[@value='Sign in']").click()

File: selenium-webdriver-junit/tests/LoginPage.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage {
    public WebDriver driver;
    public LoginPage(WebDriver driver) {
        this.driver = driver;
    }

    public void enterUsername(String username) {
        driver.findElement(By.id("username")).clear();
        driver.findElement(By.id("username")).sendKeys(username);
    }

    public void enterPassword(String password) {
        driver.findElement(By.id("password")).clear();
        driver.findElement(By.id("password")).sendKeys(password);
    }

    public void clickSignIn() {
        driver.findElement(By.xpath("//input[@value='Sign in']")).click();
    }
}
